src/core/scene.cpp

```diff
@@ -70,7 +72,11 @@ void Scene::AddActor( Actor* Object )
 
 void Scene::Update( double DeltaTime )
 {
-
+    if (mSendMapStarted)
```

Comment:
// TODO clean this up
really don't like the handle map ready counter idea, and sending the mapstarted event from scene update
i'll hate myself for suggesting this, but a state machine is rather fitting here. at least an enum SceneState instead if the sendmapstarted, counter, whatever stuff